analyzer/consensus/consensus.go

```diff
@@ -611,6 +586,124 @@ func (m *Main) queueNodeEvents(batch *storage.QueryBatch, data *storage.Registry
 	return nil
 }
 
+func (m *Main) queueRegistryEvents(batch *storage.QueryBatch, data *storage.RegistryData) error {
+	eventInsertQuery := m.qf.ConsensusEventInsertQuery()
+	backend := analyzer.BackendRegistry
+
+	for _, event := range data.Events {
+		var ty analyzer.Event
+		var body []byte
+		var err error
+
+		switch {
+		case event.RuntimeEvent != nil:
+			ty = analyzer.EventRegistryRuntime
+			body, err = json.Marshal(event.RuntimeEvent)
+
+			if err != nil {
+				return err
+			}
+		case event.EntityEvent != nil:
+			ty = analyzer.EventRegistryEntity
+			body, err = json.Marshal(event.EntityEvent)
+
+			if err != nil {
+				return err
+			}
+		case event.NodeEvent != nil:
+			ty = analyzer.EventRegistryNode
+			body, err = json.Marshal(event.NodeEvent)
+
+			if err != nil {
+				return err
+			}
+		case event.NodeUnfrozenEvent != nil:
+			ty = analyzer.EventRegistryNodeUnfrozen
+			body, err = json.Marshal(event.NodeUnfrozenEvent)
+
+			if err != nil {
+				return err
+			}
+		}
+
+		batch.Queue(eventInsertQuery,
+			backend.String(),
+			ty.String(),
+			string(body),
+			data.Height,
+			event.TxHash.Hex(),
+			0,
+		)
+	}
+
+	return nil
+}
+
+func (m *Main) prepareRootHashData(ctx context.Context, height int64, batch *storage.QueryBatch) error {
+	source, err := m.source(height)
+	if err != nil {
+		return err
+	}
+
+	data, err := source.RootHashData(ctx, height)
+	if err != nil {
+		return err
+	}
+
+	for _, f := range []func(*storage.QueryBatch, *storage.RootHashData) error{
+		m.queueRootHashEvents,
+	} {
+		if err := f(batch, data); err != nil {
+			return err
+		}
+	}
```

Comment:
Optional:
```suggestion
	if err := m.queueRootHashEvents(batch, data); err != nil {
			return err
		}
```

though I can see preferring to follow the format of the other similar funcs.